dhis-2/dhis-services/dhis-service-program-rule/src/main/java/org/hisp/dhis/programrule/engine/DefaultProgramRuleEngineService.java

```diff
@@ -82,12 +85,18 @@ public DefaultProgramRuleEngineService( ProgramRuleEngine programRuleEngine,
     public List<RuleEffect> evaluateEnrollmentAndRunEffects( long programInstanceId )
     {
         ProgramInstance programInstance = programInstanceService.getProgramInstance( programInstanceId );
-        List<RuleEffect> ruleEffects = getRuleEffects( programInstance );
+
+        if ( programInstance == null )
+        {
+            return Lists.newArrayList();
+        }
+
+        List<RuleEffect> ruleEffects = getRuleEffects( programInstance, Optional.empty(),
+            programInstance.getProgramStageInstances() );
 
         for ( RuleEffect effect : ruleEffects )
         {
-            ruleActionImplementers.stream().filter( i -> i.accept( effect.ruleAction() ) ).forEach( i ->
-            {
+            ruleActionImplementers.stream().filter( i -> i.accept( effect.ruleAction() ) ).forEach( i -> {
```

Comment:
It is duplicated but it is slightly different because the implement method is called one time for ProgramInstance and another time for ProgramStageInstance.
I will eventually refactor it to avoid this duplication, just not yet.